pkg/k8sutil/rollout.go

```diff
@@ -0,0 +1,258 @@
+package k8sutil
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	clientappsv1 "k8s.io/client-go/kubernetes/typed/apps/v1"
+	"k8s.io/client-go/tools/cache"
+	watchtools "k8s.io/client-go/tools/watch"
+)
+
+// RolloutTimeout is the timeout to wait for a daemonset or deployment to rollout changes.
+const RolloutTimeout = 15 * time.Minute
+
+// RolloutRestartDaemonSet is the programmatic equivalent of 'kubectl rollout restart'.
+func RolloutRestartDaemonSet(dsi clientappsv1.DaemonSetInterface, name string) (int64, error) {
+	ds, err := dsi.Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		return 0, fmt.Errorf("getting DaemonSet: %w", err)
+	}
+
+	if ds.Spec.Template.Annotations == nil {
+		ds.Spec.Template.Annotations = map[string]string{}
+	}
+
+	ds.Spec.Template.Annotations["kubectl.kubernetes.io/restartedAt"] = time.Now().String()
+
+	newDS, err := dsi.Update(context.TODO(), ds, metav1.UpdateOptions{})
```

Comment:
The pod spec hasn't changed, right? How does this forces pods to be re-created? Does it change the generation or something? Sorry, but looking at the [doc][link] I didn't find. Asking to make sure I follow what we are doing here :)

[link]: https://pkg.go.dev/k8s.io/client-go@v0.21.0/kubernetes/typed/apps/v1#DaemonSetInterface

Reply:
haha, it was so obvious I didn't see it!